specifications/sep-payment-channel-mechanism.md

```diff
@@ -0,0 +1,631 @@
+
+Participants can contest a close if the close is not the most recent agreed
+closing state of the payment channel.
+
+Participants can attempt to close the channel at a state that is earlier in the
+history of the channel than the most recently agreed to state. A participant who
+is a malicious actor might attempt to do this if an earlier state benefits them.
+
+The malicious participant can do this by performing the [Uncoordinated
+Close](#Uncoordinated-Close) process with a declaration transaction that is not
+the most recently signed declaration transaction.
+
+The other participant can identify that the close process has started at an
+earlier state by monitoring changes in escrow account E's sequence. If the other
+participant sees the sequence number of escrow account E change to a value that
+is not the most recently used s_i, they can use the following process to contest
+the close. A participant contests a close by submitting a more recent
+declaration transaction and closing the channel at the actual final state.
+
+1. Get E's sequence number n
+2. If s_{e+1} >= n < s_i, go to step 3, else go to step 1
+3. Submit most recent D_i
+4. Wait observation period O
+5. Submit C_i
+
+#### Changing the Channel Setup
+
+The payment channel setup can be altered with on-chain transactions after
+channel setup. Some of the operations used to alter the channel setup may fail
+even if the transactions are valid, while others will always succeed if the
+transactions are valid.
+
+Some operations are implemented in a two-step process. Participants agree on a
+new closing state at a future iteration by signing C_i and D_i transactions
+where i has skipped an iteration that is not yet executable because the D_i's
+`minSeqNum` is also set in the future. Participants then sign a transaction to
+make the change that only moves the sequence of escrow account E to satisfy the
+`minSeqNum` of the future D_i.
+
+Operations that can fail and change the balances of the channel have the
+following requirements as well:
+
+- The transaction that can fail must have its source account set to an account
+that is not escrow account E.
+- The transaction that can fail must contain a `BUMP_SEQUENCE` operation that
+bumps escrow account E's sequence number to a sequence number that makes the D_i
+executable.
+
+Operations where failure cannot occur or is of no consequence:
+
+- [Change the Observation Period](#Change-the-Observation-Period)
+- [Add Trustline](#Add-Trustline)
+- [Remove Trustline](#Remove-Trustline)
+
+Operations that can fail:
+- [Deposit by Initiator](#Deposit-by-Initiator)
+- [Deposit by Responder](#Deposit-by-Responder)
+- [Withdraw](#Withdraw)
+
+##### Add Trustline
+
+Participants can add additional trustlines if they plan to make deposits of new balances.
+
+1. I and R sign and exchange signatures for trustline transaction TA_i.
+2. I or R submit TA_i.
+
+If the remove trustline transaction TA_i fails or is never submitted, there is
```

Comment:
You probably meant **add** trustline here?

Reply:
Thanks! d4022e5